lib/preprocessor.js

```diff
@@ -36,6 +37,94 @@ var createNextProcessor = function (preprocessors, file, done) {
   }
 }
 
+/* Merge two lists, removing duplicates, and doing everything possible to
```

Comment:
I suggest we put this into its own file or actually pull it into it's own repo. This is quite complex but very useful and might help others as well when we publish it to npm. Also that makes testing it very easy :)